parsers/dortmund.py

```diff
@@ -0,0 +1,138 @@
+import re
+import datetime
+import urllib.request as rq
+
+from urllib.parse import urlencode as urlencode
+from bs4 import BeautifulSoup
+from utils import Parser
+
+from pyopenmensa.feed import LazyBuilder
+
+# dictionary for categories
+categories = {
+    103: 'Tagesgericht',
+    101: 'Menü 1',
+    102: 'Menü 2',
+    104: 'Vegetarisches Menü',
+    105: 'Aktionsteller',
+    106: 'Aktionsteller Fisch',
+    107: 'Aktionsteller Vegan',
+    108: 'Grillstation',
+    109: 'Selfservice Station',
+    115: 'Restaurant',
+    116: 'Beilage',
+    120: 'Menü 1',
+    121: 'Menü 2',
+    122: 'Menü 3',
+    123: 'Menü 3',
+    124: 'Vegetarisches Menü',
+    125: 'Vegan',
+    127: 'Aktionsteller',
+    128: 'Schnelle Theke',
+    130: 'Sonstige',
+    131: 'Low Carb Teller',
+    135: 'Pastatheke Pastaauswahl',
+    136: 'Pastatheke Toppings',
+    137: 'Pastatheke Saucen',
+}
+
+def getAndFormatPrice(price):
+    price = re.search('(\d+,\d{1,2})', price)
+    if price != None:
+        formatted = re.sub('(\d+),(\d+)', r'\1.\2', price.group(0))
+        return float(formatted)
+    else:
+        return '-' 
+
+def getWeekdays(day):
+    currentWeekday = day.weekday()
+    week = []
+    wstart = -currentWeekday
+    wend = 7 - currentWeekday
+    for d in range(wstart, wend):
+        nday = day + datetime.timedelta(days=d)
+        week.append(nday.isoformat())
+    return week
+
+def parse_url(url, today=False):
+    canteen = LazyBuilder()
+    day = datetime.date.today()
+
+    week = getWeekdays(day)
+
+    for d in range(7):
+        py = {'tx_pamensa_mensa[date]' : week[d]}
+        payload = urlencode(py).encode('ascii')
+        data = rq.urlopen(url, payload).read().decode('utf-8')
+        soup = BeautifulSoup(data, 'html.parser')
+
+        parse_day(canteen, soup, week[d])
+
+    legend_url = 'https://www.stwdo.de/mensa-co/allgemein/zusatzstoffe/'
+    canteen.legend = parse_legend(legend_url)
```

Comment:
Ok, moved the legend up.
In terms of the `canteen.extra_regex` I set it so that it now accepts 3 characters.
```
canteen.extra_regex = re.compile('\((?P<extra>[0-9a-zA-Z]{1,3}'
                                 '(?:,[0-9a-zA-Z]{1,3})*)\)', re.UNICODE)
```
That's what you meant?

Reply:
Yes, exactly. However, you have to use the `setLegendData` function to set the legend. Have you tested this? You should see all the notes being removed from the title and added as note:
```xml
<meal>
  <name>Gebackenes Seelachsfilet mit Dillcrème ,dazu Salzkartoffeln, Salat und Dessert</name>
  <note>Mit Fisch bzw. Meeresfrüchten</note>
  <note>Kinderteller</note>
  <note>Gluten aus Weizen &amp; Erzeugnisse</note>
  <note>Fisch &amp; Erzeugnisse</note>
  <note>mit Süßungsmitteln</note>
  <note>Gluten aus Gerste &amp; Erzeugnisse</note>
  <note>Eier &amp; Erzeugnisse</note>
  <note>Milch &amp; Erzeugnisse</note>
  <note>Sellerie &amp; Erzeugnisse</note>
  <price role="student">1.60</price>
  <price role="employee">2.70</price>
  <price role="other">3.20</price>
</meal>
```